File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool compare(int x, int y){
    return x > y;
}

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    
    vector<int> v(10000001);
    
    for(int i=0;i<tangerine.size();i++){
        v[tangerine[i]]++;
    }
    
    sort(v.begin(),v.end(),compare);
    
    for(int i=0;i<v.size();i++){       
        if(k <= 0){
            break;
        }
        answer++;
        k -= v[i];
    }
    
    return answer;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        
        Dictionary<int,int> tangerineDict = new Dictionary<int,int>();
        
        for(int i=0;i<tangerine.Length;i++){
            if(tangerineDict.ContainsKey(tangerine[i])){
                tangerineDict[tangerine[i]]++;
            }
            else{
                tangerineDict[tangerine[i]] = 1;
            }
        }
        
        List<int> count = new List<int>(tangerineDict.Values);
        count.Sort((a,b) => b.CompareTo(a));
        
        int sum = 0;
        
        for(int i=0;i<count.Count;i++){
            sum += count[i];
            answer++;
            
            if(sum >= k){
                break;
            }
        }
        
        return answer;
    }
}